### PR TITLE
Remove ApiSource

### PR DESCRIPTION
**Pull request by andreamlin**
- Remove `ApiSource` (enum with values `PROTO`, `DISCOVERY`) to make toolkit pipeline more general
- Abstract some `ProtoElement`s away from SurfaceNamer.java
- discogapic message surface to use same getter/setter names as gapic surface; e.g. a getter for a `List<Warnings>` will be named `getWarningsList` to follow gapic naming convention
- remove various outdated TODOs

### CODE REVIEW

File: src/main/java/com/google/api/codegen/GapicContext.java [hunk 1]
```diff
@@ -68,41 +62,9 @@ public ServiceMessages messages() {
     return serviceMessages;
   }
 
-  /** Return the name of the class which is the GAPIC wrapper for this API interface. */
```
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/GapicContext.java]: All these methods were unused.

File: src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java [hunk 2]
```diff
@@ -147,6 +148,9 @@ private StaticLangApiMessageView generateSchemaClasses(
       DiscoGapicInterfaceContext documentContext,
       Schema schema) {
 
+    // TODO(andrealin): Remove DiscoGapicNamer from DiscoveryField.create().
```
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java]: Will be fixed in https://github.com/googleapis/toolkit/pull/1864.

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/gapic/GapicParser.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.gapic;
+
+import com.google.api.tools.framework.aspects.documentation.model.DocumentationUtil;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.api.tools.framework.model.ProtoFile;
+
+/* Util class for parsing ProtoElements into canonical inputs for the gapic transformer pipeline. */
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/gapic/GapicParser.java]: Use a double star for javadocs

File: src/main/java/com/google/api/codegen/gapic/GapicParser.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/* Copyright 2017 Google LLC
```
Comment (vchudnov-g) [inline src/main/java/com/google/api/codegen/gapic/GapicParser.java]: s/2017/2018/

**Pull request merged**